src/main/java/com/google/sps/data/EmailSender.java

```diff
@@ -14,65 +14,23 @@
 
 package com.google.sps.data;
 
-// using SendGrid's Java Library
-// https://github.com/sendgrid/sendgrid-java
-import com.sendgrid.Method;
-import com.sendgrid.Request;
 import com.sendgrid.Response;
-import com.sendgrid.SendGrid;
-import com.sendgrid.helpers.mail.Mail;
 import com.sendgrid.helpers.mail.objects.Content;
 import com.sendgrid.helpers.mail.objects.Email;
 import java.io.IOException;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.util.stream.Stream;
-import java.util.Map;
 import java.util.HashMap;
-import javax.servlet.annotation.WebServlet;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import java.net.URL;
 
-// Handles sending emails.
-@WebServlet("/email")
-public class EmailSender {
-  // The email messages are being sent from.
-  private final Email sender;
-  private final SendGrid sg;
+/** EmailSender includes the basic methods anything managing sending emails must support. */
+public interface EmailSender {
 
-  public EmailSender(Email sender) throws Exception {
-    this.sender = sender;
-    this.sg =
-        new SendGrid(
-            new SecretFetcher("interview-me-step-2020").getSecretValue("SENDGRID_API_KEY"));
-  }
+  // Sends an email to the "recipient" Email, with specified subject and content. Returns a response
+  // from the sendgrid email sending service.
 
-  // Sends an email from the "from" Email to the "to" Email, with specified subject and content.
-  public Response sendEmail(Email recipient, String subject, Content content)
-      throws IOException, Exception {
-    Mail mail = new Mail(sender, subject, recipient, content);
-
-    Request request = new Request();
-    Response response;
-    request.setMethod(Method.POST);
-    request.setEndpoint("mail/send");
-    request.setBody(mail.build());
-    response = sg.api(request);
-    return response;
-  }
+  public int sendEmail(Email recipient, String subject, Content content) throws IOException;
 
   // Returns the contents of the file specified at filePath as a String. Useful for converting
-  // predifined email templates to text.
-  public static String fileContentToString(String filePath) throws IOException {
-    StringBuilder contentBuilder = new StringBuilder();
-    try (Stream<String> stream = Files.lines(Paths.get(filePath), StandardCharsets.UTF_8)) {
-      stream.forEach(s -> contentBuilder.append(s).append("\n"));
-    }
-    return contentBuilder.toString();
-  }
+  // predefined email templates to text.
+  public String fileContentToString(String filePath) throws IOException;
```

Comment:
This probably shouldn't be part of the interface (and is the root cause of the weirdness when developing the test fake). Is it possible to move this to a utils class (maybe "EmailUtils")?

Reply:
Done

Reply:
Done